mega-zep-frontend/src/main/angular/frontend/src/app/modules/monthly-report/services/monthly-report.service.spec.ts

```diff
@@ -1,17 +1,86 @@
 import {TestBed} from '@angular/core/testing';
 
 import {MonthlyReportService} from './monthly-report.service';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
-
-describe('TimeService', () => {
-  beforeEach(() => TestBed.configureTestingModule({
-    imports: [
-      HttpClientTestingModule
-    ]
-  }));
-
-  it('should be created', () => {
-    const service: MonthlyReportService = TestBed.get(MonthlyReportService);
-    expect(service).toBeTruthy();
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {MonthlyReport} from "../models/MonthlyReport";
+import {Employee} from "../../shared/models/Employee";
+import {ConfigService} from "../../shared/services/config/config.service";
+
+describe('MonthlyReportService', () => {
+
+  let monthlyReportService: MonthlyReportService;
+  let httpTestingController: HttpTestingController;
+  let configService: ConfigService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ]
+    });
+
+    monthlyReportService = TestBed.inject(MonthlyReportService);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    configService = TestBed.inject(ConfigService);
+  });
+
+  it('#should be created', () => {
+    expect(monthlyReportService).toBeTruthy();
+  });
+
+  it('#getAll - should return MonthlyReport', (done) => {
+    monthlyReportService.getAll()
+      .subscribe(monthlyReport => {
+        expect(monthlyReport).toEqual(MonthlyReportMock.monthlyReport);
+        done();
+      });
+
+    const testResult = httpTestingController.expectOne(configService.getBackendUrlWithContext('/worker/monthendreports'));
```

Comment:
Hier brauchst du dann, nach der Änderung oben, folgenden Code:
`
configServiceMock.getBackendUrlWithContext.and.return('/worker/monthendreports')
`

Somit ist die Abhängigkeit zum ConfigService entfernt und du kannst den Service für sich testen.